components/Main.js

```diff
@@ -32,6 +35,14 @@ const UI_STRINGS = {
 }
 
 export default function Main() {
+  const { user, handleLogin } = useAuthentication()
+
+  const { isInitialized } = useSecrets()
+
+  if (!user || !isInitialized) {
+    return <Auth {...{ handleLogin }} />
```

Comment:
Good question Simone! Here my reasoning was: If the user is not authenticated, then they should not be allowed to even initialize the data store. Off course behind the scenes there is nothing preventing one from happening after the other but I decided to settle this way to avoid getting stuck here while many other aspects of the implementation were incomplete.

Do you think it is okay if we initialize the secrets data together with the fonts and before the user is authenticated. I'm assuming I should've asked you this before but now that we're on a more stable state with the overall solution this change should be easy enough to make in a few minutes.

Reply:
Yes I think the second approach you describe makes more sense. I wouldn't expect the secrets initialization to overlap with the user authentication. That is, at least unless we want to tie the secret to the user, which would kinda make sense but that you're not handling here anyway and would require a more complex implementation. Let's consider secret initialization in the same way as fonts for the time being, then

Reply:
Done!